dencoding/yaml_decoder_test.go

```diff
@@ -121,16 +125,15 @@ spam:
 			got = append(got, v)
 		}
 
-		fooMap := dencoding.NewMap().
-			Set("bar", int64(1)).
-			Set("baz", "baz")
-		spamMap := dencoding.NewMap().
-			Set("ham", "eggs").
-			Set("foo", fooMap)
-
 		exp := dencoding.NewMap().
-			Set("foo", fooMap).
-			Set("spam", spamMap)
+			Set("foo", dencoding.NewMap().
+				Set("bar", int64(1)).
+				Set("baz", "baz")).
+			Set("spam", dencoding.NewMap().
+				Set("ham", "eggs").
+				Set("bar", int64(1)).
+				Set("baz", "bazbaz")).
+			Set("baz", "baz")
```

Comment:
With the new override tests, I found it easiest to not use the `fooMap` and rather be explicit about the result.